ecosystem/typescript/sdk/src/bcs/helper.ts

```diff
@@ -45,6 +45,35 @@ export function deserializeVector(deserializer: Deserializer, cls: any): any[] {
   return list;
 }
 
+// Recursive function to serialize n-dimensional arrays
+function serializeNestedArrayWithFunc(value: any[], func: string, serializer: Serializer): void {
+  // Verify value is null/undefined
+  if (value == null) {
```

Comment:
How about checking `if (!Array.isArray(value))`? This would cover `null`, `undefined` and invalid`value.length`.